File: src/es/greuze/sandbox/prisoners/GentlePrisoner.java
package es.greuze.sandbox.prisoners;

import java.io.*;
import java.net.ServerSocket;
import java.net.Socket;

public class GentlePrisoner extends Prisoner {

    @Override
    public String giveAnswer() {
        return "SILENT";
    }

    @Override
    public void notifyPoliceResponse(String response) {
        System.out.println("GentlePrisoner got response '" + response + "' from Police");
    }

    public static void main(String[] args) {
        GentlePrisoner prisoner = new GentlePrisoner();
        prisoner.initialize(5000);
        prisoner.startContest();
    }
}

File: scala/src/GentlePrisoner.scala
package es.greuze.sandbox.prisoners;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.io.PrintWriter;
import java.net.ServerSocket;

class GentlePrisoner extends es.greuze.sandbox.prisoners.Prisoner {
    def giveAnswer() : String = {
        "SILENT"
    }

    def notifyPoliceResponse(response: String) {
        println("Police response: " + response)
    }
}

object PrisonerClient {

    def main(args : Array[String]) {
        val p = new GentlePrisoner
        p.initialize(5001)
        p.startContest()
    }
}
